### PR TITLE
Remove references to TaskInfo from StatementResource

### PR DESCRIPTION
**Pull request by nileema**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/QueryExecution.java [hunk 1]
```diff
@@ -51,6 +54,11 @@ Duration waitForStateChange(QueryState currentState, Duration maxWait)
 
     void addStateChangeListener(StateChangeListener<QueryState> stateChangeListener);
 
+    /**
+     * @return Exchange URIs for the root stage, empty set if root stage buffers are not yet added
+     */
+    Set<URI> getRootExchangeLocations();
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/QueryExecution.java]: Make this `Optional<Set<URI>>` and then empty means they are not set.

File: presto-main/src/main/java/com/facebook/presto/execution/QueryManager.java [hunk 2]
```diff
@@ -37,4 +39,9 @@ Duration waitForStateChange(QueryId queryId, QueryState currentState, Duration m
     void cancelQuery(QueryId queryId);
 
     void cancelStage(StageId stageId);
+
+    /**
+     * @return Exchange URIs for the root stage, empty set if root stage buffers are not yet added
+     */
+    Set<URI> getRootExchangeLocations(QueryId queryId);
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/QueryManager.java]: same here

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java [hunk 3]
```diff
@@ -138,6 +146,19 @@ public SqlQueryScheduler(QueryStateMachine queryStateMachine,
         this.executor = executor;
 
         rootStage.addStateChangeListener(state -> {
+            if (state != PLANNED && state != SCHEDULING) {
+                // We need all the exchange locations for the root stage to be available in one shot.
+                // If the stage is past the PLANNING and SCHEDULING state, all tasks are added and
+                // there won't be any new locations.
+                ImmutableSet.Builder<URI> uris = ImmutableSet.<URI>builder();
+                for (TaskInfo taskInfo : rootStage.getStageInfo().getTasks()) {
+                    List<BufferInfo> buffers = taskInfo.getOutputBuffers().getBuffers();
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java]: IIRC there should only be one buffer per task for the output stage.  If so, we should enforce that here.

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java [hunk 4]
```diff
@@ -138,6 +146,19 @@ public SqlQueryScheduler(QueryStateMachine queryStateMachine,
...
+                    for (BufferInfo buffer : buffers) {
+                        uris.add(uriBuilderFrom(taskInfo.getSelf()).appendPath("results").appendPath(buffer.getBufferId().toString()).build());
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java]: I thought there was a utility method for this?

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java [hunk 5]
```diff
@@ -82,7 +87,10 @@
     private final ExecutionPolicy executionPolicy;
     private final Map<StageId, SqlStageExecution> stages;
     private final ExecutorService executor;
+
     private final StageId rootStageId;
+    private final Set<URI> rootExchangeLocations = new HashSet<>();
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java]: This would also be optional

Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java]: BTW, this field is mutated in a callback.  Do you need a lock?

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java [hunk 6]
```diff
@@ -396,6 +417,11 @@ public void abort()
         }
     }
 
+    public Set<URI> getRootExchangeLocations()
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java]: Does this need to be synchronized?

**Comment by dain** [comment]
There are build failures

**Comment by dain** [comment]
This is still seeing test failures

**Comment by nileema** [comment]
I realized I don't need this. Closing this PR.

**Pull request closed without merging**